src/policy/fees.cpp

```diff
@@ -649,6 +649,23 @@ void CBlockPolicyEstimator::processBlock(unsigned int nBlockHeight,
     shortStats->UpdateMovingAverages();
     longStats->UpdateMovingAverages();
 
+    // Do not consider transactions that were CPFP'd.
+    // For each mempool transaction that was included in a block, if its individual feerate is higher
+    // than its ancestor score it incentivized the inclusion of some of its parents in the block.
+    // Remove all such parents whose feerate is lower than the child's of the tracking before updating
+    // the data points.
+    for (const auto& entry : entries) {
+        CFeeRate individual_feerate{entry->GetFee(), static_cast<uint32_t>(entry->GetTxSize())};
+        CFeeRate ancestor_score{entry->GetModFeesWithAncestors(), static_cast<uint32_t>(entry->GetSizeWithAncestors())};
+
+        if (individual_feerate > ancestor_score) {
+            for (const CTxMemPoolEntry& parent : entry->GetMemPoolParents()) {
+                CFeeRate parent_feerate{parent.GetFee(), static_cast<uint32_t>(parent.GetTxSize())};
+                if (parent_feerate < individual_feerate) _removeTx(parent.GetTx().GetHash(), /* inBlock = */true);
```

Comment:
```c++
 if (parent_feerate < individual_feerate) _removeTx(parent.GetTx().GetHash(), /* inBlock = */true);
```
This means parents that are confirmed from previous blocks are going to be removed as well right?  not just parents from this block we are processing?

Reply:
No, any parents from previous blocks would not exist in the mempool anymore. They would have been removed when that block was processed.

Reply:
Thank you
If the ancestors are confirmed in the previous block, and the child (with fee rate higher than ancestors score) is in the block we are processing, is that also considered CPFP?  if yes then we are not removing those ancestors here right?

Also why is  `/* inBlock = */true`  ?

Reply:
CPFP refers to a couple of unconfirmed transactions where the descendant transaction pays a higher feerate than the parent transaction. If the parent transaction is already confirmed, it’s no longer a CPFP situation.

Reply:
Thank you @Xekyo

Reply:
> Also why is /* inBlock = */true ?

That's so it's not counted as a failure.